libraries/tock-register-interface/src/macros.rs

```diff
@@ -119,12 +119,17 @@ macro_rules! register_bitmasks {
 #[macro_export]
 macro_rules! register_bitfields {
     {
-        $valtype:ty, $( $(#[$inner:meta])* $reg:ident $fields:tt ),*
+        $valtype:ty, $( $(#[$inner:meta])* $vis:vis $reg:ident $fields:tt ),*
     } => {
         $(
             #[allow(non_snake_case)]
             $(#[$inner])*
-            pub mod $reg {
+            $vis mod $reg {
+                // Visibility note: This is left always `pub` as it is not
+                // meaningful to restrict access to the `Register` element of
+                // the register module if the module itself is in scope
+                //
+                // (if you can access $reg, you can access $reg::Register)
                 #[derive(Clone, Copy)]
                 pub struct Register;
```

Comment:
Why not just put `$vis` here (instead of `pub`) and remove the need for the comment? Or would that not work?

Reply:
You don't want `$vis` here. If the bitfield is `pub` then you'd see no change. However if the bitfield is not `pub` then if this was also `$vis` each of the members of the bitfield would also not be accessible, i.e. you'd have to write:

```rust
register_bitfields![u32,
    [nopubhere] mcause [
        // only support 32 bit targets
        pub isException OFFSET(31) NUMBITS(1) [],
        pub reason OFFSET(0) NUMBITS(31) []     // <-- must grant explicit pub to each field, o/w inaccessible
    ]
];